src/dolphin/os/OSExi.c

```diff
@@ -111,17 +114,21 @@ BOOL EXIImmEx(EXIChannel chan, void* buf, s32 len, u32 mode)
 {
     while (len) {
         s32 xLen = (len < 4) ? len : 4;
-        if (!EXIImm(chan, buf, xLen, mode, NULL)) {
+
+        if (!EXIImm(chan, buf, xLen, mode, NULL))
             return FALSE;
-        }
 
-        if (!EXISync(chan)) {
+        if (!EXISync(chan))
             return FALSE;
-        }
 
+/// @todo what
+#if defined(__MWERKS__) && defined(GEKKO)
         (u8*) buf += xLen;
+#endif
```

Comment:
```suggestion
        buf = (u8*) buf + xLen;
```

Reply:
Did you compile this? I think I tried it.

Reply:
Yeah it works

Reply:
It also just does nothing as far as I can tell...

Reply:
It offsets buf by xLen bytes for the next iteration.

Reply:
Oh right, duh.